src/listener.js

```diff
@@ -1,183 +1,234 @@
 'use strict'
 
-const LP = require('./rpc/lp')
-const pull = require('pull-stream/pull')
-const handshake = require('pull-handshake')
-const { JoinInit, JoinChallenge, JoinChallengeSolution, JoinVerify, Discovery, DialRequest, DialResponse, ErrorTranslations } = require('./rpc/proto')
-
-const prom = (f) => new Promise((resolve, reject) => f((err, res) => err ? reject(err) : resolve(res)))
+const debug = require('debug')
+const log = debug('libp2p:stardust:listener')
+log.error = debug('libp2p:stardust:listener:error')
 
-const sha5 = (data) => crypto.createHash('sha512').update(data).digest()
+const EventEmitter = require('events')
 
 const crypto = require('crypto')
-const ID = require('peer-id')
+const PeerId = require('peer-id')
 const PeerInfo = require('peer-info')
+const toConnection = require('libp2p-utils/src/stream-to-ma-conn')
 
-const debug = require('debug')
-const log = debug('libp2p:stardust:client')
+const Wrap = require('it-pb-rpc')
+const { int32BEDecode, int32BEEncode } = require('it-length-prefixed')
+const { JoinInit, JoinChallenge, JoinChallengeSolution, JoinVerify, Discovery, DiscoveryAck, DialRequest, DialResponse, ErrorTranslations } = require('./proto')
 
-const EventEmitter = require('events').EventEmitter
+const sha5 = (data) => crypto.createHash('sha512').update(data).digest()
 
 function translateAndThrow (eCode) {
   throw new Error(ErrorTranslations[eCode] || ('Unknown error #' + eCode + '! Please upgrade libp2p-stardust to the latest version!'))
 }
 
-function noop () { }
-
-const ACK = Buffer.from('01', 'hex')
-
+/**
+* Stardust Transport Listener
+* @class
+*/
 class Listener extends EventEmitter {
-  constructor (client, handler) {
+  /**
+   * @constructor
+   * @param {Object} properties - properties for the listener
+   * @param {function (Connection)} properties.handler - New connection handler
```

Comment:
handler should be optional
```suggestion
   * @param {function (Connection)} [properties.handler] - New connection handler
```